src/main_chart.py

```diff
@@ -103,23 +103,42 @@ def get_main_chart_figure(
     color_map_liquidity = {"debt_token_supply": "#1f77b4"} # blue
     figure = plotly.graph_objs.Figure()
 
+    custom_columns = [  
+        "liquidable_debt",
+        "liquidable_debt_zkLend",
+        "liquidable_debt_at_interval_Nostra Alpha",
+        "liquidable_debt_at_interval_Nostra Mainnet"
+    ]
+    customdata = []
+    for col in custom_columns:
+        if col in data.columns:
+            customdata.append(data[col].values)
+        else:
+            customdata.append([0] * len(data)) # Use 0 if the column is missing
```

Comment:
move `len(data)` to constant. the length of data will be fix

Reply:
Please ser could you explain a bit more what you mean??

Reply:
@Dprof-in-tech I think you didn't commit this function `get_custom_data` so I can't help you until you push all your changes.
BTW:
instead of using every time this: `len(data)`
you can do like this:

data_length = len(data)

customdata.append([0] * data_length) 
and in another places where are you using `len(data)`

Reply:
just fixed all ser